support/cas-server-support-pac4j-core-clients/src/main/java/org/apereo/cas/support/pac4j/logout/TriggerCasSLOLogoutHandler.java

```diff
@@ -0,0 +1,92 @@
+package org.apereo.cas.support.pac4j.logout;
+
+import org.apereo.cas.ticket.Ticket;
+import org.apereo.cas.ticket.TicketGrantingTicketImpl;
+import org.apereo.cas.ticket.factory.DelegatedTicketGrantingTicketFactory;
+import org.apereo.cas.ticket.registry.TicketRegistry;
+import org.apereo.cas.web.cookie.CasCookieBuilder;
+import org.apereo.cas.web.flow.CasWebflowConstants;
+import org.apereo.cas.web.support.WebUtils;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.apache.commons.lang3.StringUtils;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.context.session.SessionStore;
+import org.pac4j.core.logout.handler.LogoutHandler;
+import org.pac4j.core.profile.factory.ProfileManagerFactory;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.webflow.context.servlet.ServletExternalContext;
+import org.springframework.webflow.execution.Action;
+import org.springframework.webflow.execution.RequestContextHolder;
+
+import jakarta.servlet.http.HttpServletRequest;
+import java.util.function.Predicate;
+
+/**
+ * The pac4j logout handler which triggers a CAS logout.
+ *
+ * @author Jerome LELEU
+ * @since 7.0.0
+ */
+@Slf4j
+@RequiredArgsConstructor
+public class TriggerCasSLOLogoutHandler implements LogoutHandler {
+
+    private final CasCookieBuilder ticketGrantingTicketCookieGenerator;
+
+    private final TicketRegistry ticketRegistry;
+
+    private final ConfigurableApplicationContext applicationContext;
+
+    @Override
+    public void recordSession(final WebContext context, final SessionStore sessionStore, final String key) {
+        context.setRequestAttribute(DelegatedTicketGrantingTicketFactory.DELEGATED_SESSION_KEY_REQUEST_ATTRIBUTE, key);
```

Comment:
When pac4j says that a delegated authentication must be registered for a key, I add it as a request attribute and retrieve this request attribute in a specific `DelegatedTicketGrantingTicketFactory.java` (used when the authn delegation is enabled).